test/client_test.cc

```diff
@@ -39,7 +39,7 @@ TEST_F(ClientTest, AutoConcurrencyRun) {
   argv.push_back("error");
   argv.push_back("http://localhost:63657/");
   Main program(argv.size(), argv.data());
-  EXPECT_TRUE(program.run());
+  EXPECT_FALSE(program.run());
```

Comment:
Why false?

Reply:
Ah, yes, this surely deserves some explanation. 

So the c++ tests talk to a non-existing server. We've gone through some trouble with the c++ integration test server because of issues with singleton runtimes in the past.

Before this change, NH didn't care about that there was no server listening, but now that we have these default termination predicates it marks the execution as a failure, because connection failures are observed. That's why this line has `EXPECT_FALSE`.

In https://github.com/envoyproxy/nighthawk/pull/137 we wrap the grpc service in the python integration tests.
That takes us one step closer towards porting these tests to that, which would be helpful because then we can easily spin up test servers again. 

(this also applies to some of your comments below)

Reply:
OK.  We probably want to get c++ integration tests working in general, but I know that difficult.  Do we have a core Envoy issue for this?  Also could we add a comment?

Reply:
We played whack-a-mole for a while as the issues would jump around as dependencies on the singleton where moved/introduced in the Envoy code-base. So we went through various hoops to work around (including using fork() to dish out a new process to run the actual tests), each attempt introduced new kinds of trouble, and finally we gave up. There is no core Envoy issue for this that I know of.

I'll add a comment here to make note of why we expect these failures, makes sense